controllers/comments.controllers.js

```diff
@@ -1,11 +1,34 @@
-const { fetchCommentsByArticleId } = require('../models/comments.models');
+const { checkArticleExists } = require('../models/articles.models');
+const {
+  fetchCommentsByArticleId,
+  addCommentByArticleId,
+} = require('../models/comments.models');
 
 exports.getCommentsByArticleId = (req, res, next) => {
   const { article_id } = req.params;
+  const promiseArr = [fetchCommentsByArticleId(article_id)];
 
-  fetchCommentsByArticleId(article_id)
-    .then((comments) => {
+  if (article_id) {
+    promiseArr.push(checkArticleExists(article_id));
+  }
+
+  Promise.all(promiseArr)
+    .then(([comments]) => {
```

Comment:
Nice refactor after the seminar yesterday, great to see the learning being applied 🥇